src/Statistics.jl

```diff
@@ -107,6 +107,8 @@ mean(f, A::AbstractArray; dims=:) = _mean(f, A, dims)
     mean!(r, v)
 
 Compute the mean of `v` over the singleton dimensions of `r`, and write results to `r`.
+Note that since the mean! function is intended to operate without making any allocations,
+the target should not alias with the source.
```

Comment:
I think it's fine not to go into explaining why this is the case and just state
```suggestion
Note that the target must not alias with the source.
```